### DISCUSSION TITLE
Same app with multiple domains on Zeit Now

### DISCUSSION THREAD

**Discussion by tiwac100** [vercel/next.js#11200] [Help]
My use case is this: I have one app that serves multiple domains. The app fetches different data based on the domain (for example translations and CMS pages).

Currently I need to use `getInitialProps` or `getServerSideProps` because that's the only place where the `req.headers.host` is available. But theoretically all this information can be provided at build time via the correct environment as translations and pages don't change that often and don't need to be fetched on every request so that all pages could be generated statically.

Some ideas that came to mind:
- Using different branches and deploying each branch as a separate app (very error prone as each branch has to be kept up-to-date)
- Using multiple build commands in `package.json` (don't know how to use this with Zeit Now)

But all this feels not quite right. Is there a go-to route for a project structure like this?

**Comment by paulogdm** [comment]
Hello @timwachsener :wave: ,

I believe the following workflow would solve your use-case:
- Multiple projects in your ZEIT Now account.
- One GitHub repository linked to those projects.
- Different [build commands](https://zeit.co/blog/advanced-project-settings) for each project.

The team is working on these solutions as we speak. If you need a solution now, let me know so I can edit my answer with a workaround.

**EDIT**:
Use the CLI w/ GitHub actions:
```
NOW_ORG_ID=XXX NOW_PROJECT_ID=xxx now --prod -A now.staging.json
NOW_ORG_ID=YYY NOW_PROJECT_ID=yyy now --prod -A now.prod.json
```
This snippet will be unnecessary once we release "multi-env" support.

**Reply by tiwac100** [reply]
Thanks for the reply, can you provide the workaround you mentioned?

I think the only thing really missing are project-level secrets so I can use the same repository for multiple projects and define different environment vars per project.

**Reply by flutterforsale** [reply]
Hello @paulogdm 

I am also looking solution for my project. My project same https://github.com/zeit/static-fun 

How i can using Export static sites feature or getStaticProps for per subdomain?

Thanks @paulogdm

**Reply by paulogdm** [reply]
@timwachsener Edited my answer w/ the workaround.

**Reply by paulogdm** [reply]
@flutterforsale Please create a new discussion on `zeit/now`.

**Reply by revskill10** [reply]
@paulogdm When i import new project with an existing github, it says the github is already used for another project.

**Comment by tiwac100** [comment]
@paulogdm Sounds quite straightforward actually. I guess it would also be possible to use the same build command and just define different environment vars per project.

Is there an ETA? Would be great to test this as the project is in the process of being finished.

**Reply by paulogdm** [reply]
Correct. I don't have an ETA, unfortunately...
